enhancements/olm-managed-metrics.md

```diff
@@ -0,0 +1,84 @@
+
+For the uninitiated, Red Hat uses metrics collected by Telemeter to:
+* Proactively reach out to customers with "problem clusters"
+* Troubleshoot customer issues
+* Understand how individual applications are behaving at a wholistic level
+
+## Motivation
+
+OLM managed operators would significantly benefit from the ability to interact with Prometheus operators deployed on cluster. In particular, it is very difficult for Red Hat OLM managed operators to report metrics to Telemeter, which places these teams at a severe disadvantage when diagnosing customer issues.
+
+It is imparative that OLM provide these teams with the means to interact with Prometheus operators on cluster and by extension report metrics to Telemeter.
+
+### Goals
+
+* Provide operators with the means to expose metrics to prometheus via `ServiceMonitors`
+* Provide operators with the means to create `PrometheusRules`
+* Ensure that only approved Red Hat operators are scraped by Telemeter
+
+### Non-Goals
+
+* Provide generic metrics about OLM managed operators
+
+## Proposal
+
+### Bundle ServiceMonitor and PrometheusRule Objects with a CSV
+
+The bundle object should be updated to support Prometheus `ServiceMonitor` and `PrometheusRule` objects as it does with RBAC. When these objects are packaged with a CSV, OLM will be updated to deploy them along with the operator.
+
+### OLM Telemeter Support
+
+#### Telemeter Requirements
+
+Once OLM supports bundling `ServiceMonitor` and `PrometheusRule` objects with the CSV, OLM must enable Red Hat owned operators to take advantage of Telemeter. Before moving forward, let's review the requirements that must be met for Telemeter to scrape an application's metrics:
+1. The application must be deployed in a namespace that is both prefixed with `openshift-` and labbeled with`openshift.io/cluster-monitoring=true`
+2. A [ServiceMonitor object](https://github.com/coreos/prometheus-operator/blob/master/Documentation/user-guides/getting-started.md#related-resources) must be created in the namespace mentioned above and point to the application's metrics endpoint
+3. A role/rolebinding must be created that provides the prometheus service account with appropriate RBAC priviledges to discover the ServiceMonitor object in the newly created namespace.
+
+Requirement 2 is fully met in [#Bundle ServiceMonitor/PrometheusRule Objects with a CSV](#bundle-serviceMonitor-prometheusRule-objects-with-a-csv), leaving our team with the duty to fulfill requirements 1 and 3.
+
+#### Allow operator author to define a specific namespace
+
+The first step towards allowing OLM managed operators to expose metrics to Telemeter involves deploying the operator in a namespace defined by the operator author. This can be accomplished by updating the `StrategyDetailsDeployment` type to include an optional field called `namespace` that defines where the operator's namespaced resources must be deployed. 
```

Comment:
Should we have a more future proofed solution here? Rather than just defining the namespace in this field, should we also have a way of defining the namespace that an operator needs to be deployed in general? Aren't there other use cases aside from this that we should be supporting around defining this namespace?

Reply:
There is no reason this field couldn't be used to define the namespace that all operator namespaced resources need to be installed in. If you want, we could make this a top-level object in the CSV.

Reply:
@kevinrizza and I spoke offline where he suggested adding a `namespace string` field. See updated proposal.